File: util/booru_utils.py
import os.path
from json import load

# Load boorus from json
with open(os.path.dirname(__file__) + '/../booru-list.json', 'r') as f:
    booru_list = load(f)


def find_boorus_with_value_by_key(value, key='short', _list=booru_list):

    values = list(filter(lambda x: x[key] == value, _list))

    if not values:
        return None

    return values


def remove_boorus_with_values_by_key(value_list, key='short', _list=booru_list):

    values = list(
        filter(lambda x: x[key] not in value_list, _list))

    if not values:
        return None

    return values


# Some presets
booru_list_nsfw = find_boorus_with_value_by_key(True, 'nsfw', booru_list)


# Exported values
# - booru_list
# - booru_list_nsfw

# Exported functions
# - find_boorus_with_value_by_key
# - remove_boorus_with_values_by_key


# Example usage
# print(booru_list_nsfw, len(booru_list_nsfw))
# print(find_boorus_with_value_by_key('xxx', 'short'))
# print(remove_boorus_with_values_by_key(["xxx"], 'short', booru_list_nsfw))

File: util/BooruUtils.js
import booruList from '../booru-list.json';
import booruCompleteList from '../booru-complete-list.json';
import booruTypeList from '../booru-type-list.json';

function findBoorusWithValueByKey(value, key = 'domain', list = booruList) {
  return list.filter((booru) => booru[key] === value);
}

function removeBoorusWithValuesByKey(
  valueArray,
  key = 'domain',
  list = booruList
) {
  return list.filter((booru) => !valueArray.includes(booru[key]));
}

/* --- Example presets --- */
// const booruListNSFW = findBoorusWithValueByKey(true, 'nsfw')

/* --- Example usage --- */
// console.log(booruListNSFW, booruListNSFW.length)
// console.log(findBoorusWithValueByKey('rule34.xxx', 'domain'))
// console.log(removeBoorusWithValuesByKey(['rule34.xxx', 'rule34.paheal.net'], 'domain', booruListNSFW))

export {
  booruList,
  booruCompleteList,
  booruTypeList,
  findBoorusWithValueByKey,
  removeBoorusWithValuesByKey,
};
